src/app/util/attribute-storage.cpp

```diff
@@ -1462,3 +1461,48 @@ DataVersion * emberAfDataVersionStorage(const chip::app::ConcreteClusterPath & a
 
     return ep.dataVersions + clusterIndex;
 }
+
+namespace {
+class GlobalInteractionModelEngineChangedpathListener : public AttributesChangedListener
+{
+public:
+    ~GlobalInteractionModelEngineChangedpathListener() = default;
+
+    void MarkDirty(const AttributePathParams & path) override
+    {
+        InteractionModelEngine::GetInstance()->GetReportingEngine().SetDirty(path);
+    }
+};
+
+} // namespace
+
+AttributesChangedListener * emberAfGlobalInteractionModelAttributesChangedListener()
+{
+    static GlobalInteractionModelEngineChangedpathListener listener;
+    return &listener;
+}
+
+void emberAfAttributeChanged(EndpointId endpoint, ClusterId clusterId, AttributeId attributeId,
+                             AttributesChangedListener * listener)
+{
+    // Increase cluster data path
+    DataVersion * version = emberAfDataVersionStorage(ConcreteClusterPath(endpoint, clusterId));
+    if (version == nullptr)
+    {
+        ChipLogError(DataManagement, "Endpoint %x, Cluster " ChipLogFormatMEI " not found in IncreaseClusterDataVersion!", endpoint,
+                     ChipLogValueMEI(clusterId));
+    }
+    else
+    {
+        (*(version))++;
+        ChipLogDetail(DataManagement, "Endpoint %x, Cluster " ChipLogFormatMEI " update version to %" PRIx32, endpoint,
```

Comment:
This seems like it would be quite verbose...

Reply:
This is where this lived before: https://github.com/project-chip/connectedhomeip/blob/master/src/app/reporting/reporting.cpp#L40